src/Components/DrillAnimation.js

```diff
@@ -1,29 +1,245 @@
+  };
+
+  const displayNextStep = () => {
+    if (count + 1 === props.drill.steps.length) {
+      return (
+        <View style={styles.infoWrapper}>
+          <View style={styles.description}>
+            <View style={styles.wrapperFinish}>
+              <Text style={styles.fitnessNext}>Finish</Text>
+            </View>
+          </View>
+          <View style={styles.lines} />
+        </View>
+      );
+    } else {
+      return (
+        <View style={styles.infoWrapper}>
+          <View style={styles.description}>
+            <View style={styles.subWrapper}>
+              <Text style={styles.fitnessNext}>{props.drill.steps[count + 1].count}</Text>
+            </View>
+            <View style={styles.subSubWrapper}>
+              <Text style={styles.fitnessNext}>{props.drill.steps[count + 1].title}</Text>
+            </View>
+            <View style={styles.fakeWrapper} />
+          </View>
+          <View style={styles.lines} />
+        </View>
+      );
+    }
+  };
+
+  const checkSwitch = () => {
+    if (props.drill.steps && count === props.drill.steps.length) {
+      return displayFinish();
+    } else {
+      switch (props.drill.steps && props.drill.steps[count].source) {
+        case 'animation':
+          return displayAnimation();
+
+        case 'youtube':
+          return displayYoutube();
+
+        case 'vimeo':
+          return displayVimeo();
+
+        default:
+          return <Text>No visual content for this drill</Text>;
+      }
+    }
+  };
+
+  const displayFinish = () => {
+    return (
+      <View>
+        <View style={styles.containerFinish}>
+          <View style={styles.infoWrapper}>
+            <TouchableOpacity style={styles.buttonFinish} onPress={() => incrementCount()}>
+              <LinearGradient
+                style={styles.gradient}
+                colors={['#08AEEA', '#2AF598']}
+                start={{ x: 1, y: 1 }}
+                end={{ x: 0, y: 0 }}
+              >
+                <Text style={styles.textAgain}>Do it again</Text>
+              </LinearGradient>
+            </TouchableOpacity>
+          </View>
+        </View>
+      </View>
+    );
+  };
+
+  const displayAnimation = () => {
+    return (
+      <ScrollView style={styles.scrollView}>
+        <View style={styles.pageAnimation}>
+          <Animation animation={props.drill.steps[count].link} />
+        </View>
+        <View style={styles.containerAnimation}>
+          <Text style={styles.fitness}>{props.drill.steps[count].count}</Text>
+          <TouchableOpacity style={styles.buttonNext} onPress={() => incrementCount()} />
+        </View>
+        <View style={styles.subSubWrapper}>
+          <Text style={styles.instruction}>{props.drill.steps[count].instruction}</Text>
+        </View>
+      </ScrollView>
+    );
+  };
+
+  const displayYoutube = () => {
+    return (
+      <WebView
+        source={{
+          uri: props.drill.steps[count].link,
```

Comment:
What do you think of storing the youtube ID of the video in the step instead of full url ? And call the full URL in this displayYoutube function.
Depending on the used url (for example if embed option in included in url), the video display can be different, I think we want to be sure the display is always the same.

Reply:
It could be a nice idea.
Depends on how we store the data indeed

Reply:
I like it : the data would be responsible for "which video" and the component for "how to display it"

knowing the full url is robably more about "how"

Reply:
Yep, I'll suggest to do it in another PR